src/software/ai/navigator/path_manager/path_manager.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+#include <vector>
+
+#include "software/ai/navigator/obstacle/obstacle.h"
+#include "software/ai/navigator/path_planner/path_planner.h"
+#include "software/geom/point.h"
+#include "software/geom/rectangle.h"
+#include "software/geom/spline.h"
+
+/**
+ * PathManager is an interface for a path manager that,
+ * given a list of start and destination point pairs,
+ * a list of static obstacles, and navigable_area will return
+ * a list of 'best' paths between each of the pairs of
+ * start and destination
+ */
+
+using Path = std::optional<Spline>;
+
+// Used to plan a path for a robot
+class PathObjective
+{
+   public:
+    PathObjective(const Point start, const Point end, const double current_velocity,
+                  const std::vector<Obstacle> &avoid_area_obstacles)
+        : start(start),
+          end(end),
+          current_velocity(current_velocity),
+          avoid_area_obstacles(avoid_area_obstacles)
+    {
+    }
+
+    const Point start;
+    const Point end;
+    const double current_velocity;
+    const std::vector<Obstacle> avoid_area_obstacles;
```

Comment:
Can we name these `objective_specific_obstacles` instead, with a comment to the effect of  `// obstacles specific to this objective`)? I don't think it's really relevant where these came from?

Reply:
done